packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -42,50 +42,351 @@ FeatureHelper.describeFeature(module, {
-    logger,
-  });
-  const b = BFlowStage.create({
-    prerequisite: collectOrderFeedUpdate,
-    logger,
-    requestHelper,
-  });
+  // const { fetchOpportunities, c1, c2 } = FlowStageUtils.createFlow([
+  //   // { name: 'fetchOpportunities', factory: FetchOpportunitiesFlowStage.create, args: { orderItemCriteriaList }},
+  //   ['fetchOpportunities', FetchOpportunitiesFlowStage.create, { orderItemCriteriaList }],
+  //   ['c1', C1FlowStage.create],
+  //   ['c2', C2FlowStage.create],
+  //   ['p', PFlowStage.create],
+  //   ['simulateSellerApproval', TestInterfaceActionFlowStage.create, ({ p }) => ({
+  //     testName: 'Simulate Seller Approval (Test Interface Action)',
+  //     createActionFn: () => ({
+  //       type: 'test:SellerAcceptOrderProposalSimulateAction',
+  //       objectType: 'OrderProposal',
+  //       objectId: p.getResponse().body['@id'],
+  //     }),
+  //   })],
+  //   ['orderFeedUpdate', 'OrderFeedUpdate', {
+  //     testName: 'Order Feed Update (after Simulate Seller Approval)',
+  //   }],
+  //   ['b', BFlowStage.create],
+  // ], { logger, requestHelper });
+
+  // # Examples
+  //
+  // ## 1. Simple
+  // ## 1.1. With no magic at all
+  {
+    const uuid = createUuid();
+    const sellerId = getSellerId();
+    const fetchOpportunities = FetchOpportunitiesFlowStage.create({
+      orderItemCriteriaList,
+      requestHelper,
+      logger,
+    });
+    const c1 = C1FlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+      }),
+      prerequisite: fetchOpportunities,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const c2 = C2FlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+      }),
+      prerequisite: c1,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const p = PFlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+        totalPaymentDue: c2.getOutput().totalPaymentDue,
+      }),
+      prerequisite: c2,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const listenForOrderFeedUpdate = OrderFeedUpdateFlowStage.createListener({
+      prerequisite: p,
+      uuid,
+      requestHelper,
+    });
+    const simulateSellerApproval = TestInterfaceActionFlowStage.create({
+      testName: 'Simulate Seller Approval (Test Interface Action)',
+      prerequisite: listenForOrderFeedUpdate,
+      createActionFn: () => ({
+        type: 'test:SellerAcceptOrderProposalSimulateAction',
+        objectType: 'OrderProposal',
+        objectId: p.getOutput().orderId,
+      }),
+      requestHelper,
+    });
+    const collectOrderFeedUpdate = OrderFeedUpdateFlowStage.createCollector({
+      getInputState: () => ({
+        getOrderFromOrderFeedPromise: listenForOrderFeedUpdate.getOutput().getOrderFromOrderFeedPromise,
```

Comment:
Also if merging states, and having two collectors created, they would overwrite each other

Reply:
@nickevansuk I'm not quite sure what you mean. In what way would they overwrite each other?

```js
const listenForOrderFeedUpdate = ...
const someOtherStage = ...
const collectOrderFeedUpdate1 = OrderFeedUpdateFlowStage.createCollector({
  // gets getOrderFromOrderFeedPromise from listenForOrderFeedUpdate
  getInputState: FlowStageUtils.getMergedInputState(someOtherStage),
  ...
})
const collectOrderFeedUpdate2 = OrderFeedUpdateFlowStage.createCollector({
  // gets getOrderFromOrderFeedPromise from listenForOrderFeedUpdate.
  // it would then `await` the promise, as does collectOrderFeedUpdate1, which
  // will then work in the same way
  getInputState: FlowStageUtils.getMergedInputState(collectOrderFeedUpdate1),
  ...
})
```